crud/src/main/java/com/redhat/lightblue/assoc/QueryPlanChooser.java

```diff
@@ -184,34 +181,44 @@ private void iterateReferences(CompositeMetadata root,List<Conjunct> unassignedC
      * nodes. These are fixed assignments, and don't change from one
      * query plan to the other
      */
-    private void assignQueriesToPlanNodesAndEdges(List<Conjunct> queries,List<Conjunct> unassigned) {
+    private void assignQueriesToPlanNodesAndEdges(List<Conjunct> queries,
+                                                  List<Conjunct> unassigned) {
         Error.push("assignQueriesToPlanNodesAndEdges");
         LOGGER.debug("Assigning queries to query plan nodes and edges");
         try {
             for(Conjunct c:queries) {
-                Set<QueryPlanNode> nodes=c.getReferredNodes();
+                Set<CompositeMetadata> entities=c.getEntities();
                 LOGGER.debug("Conjunct {}",c);
-                switch(nodes.size()) {
+                switch(entities.size()) {
                 case 1:
                     // All query clauses that depend on fields from a single
                     // entity can be assigned to a query plan node. If clauses
                     // depend on multiple entities, their assignments may change
                     // based on the query plan.
                     LOGGER.debug("Conjunct has one entity");
-                    nodes.iterator().next().getData().getConjuncts().add(c);
+                    // Rewrite the query for that node
+                    QueryPlanNode node=qplan.getNode(entities.iterator().next());
+                    RewriteQuery rw=new RewriteQuery(compositeMetadata,node.getMetadata());
+                    QueryExpression q=rw.rewriteQuery(c.getClause(),c.getFieldInfo()).query;
+                    AnalyzeQuery analyzer=new AnalyzeQuery(node.getMetadata(),c.getReference());
+                    analyzer.iterate(q);
+                    node.getData().getConjuncts().add(new Conjunct(q,
+                                                                   analyzer.getFieldInfo(),
+                                                                   c.getReference()));
                     break;
 
                 case 2:
                     // There are two or more entities referred to in the conjunct
                     // This clause can be associated with an edge
```

Comment:
Is the intention that if there are more than 2 entities, that the default option will be used?